clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy

```diff
@@ -204,16 +210,48 @@ class BasicGoogleDeployHandler implements DeployHandler<BasicGoogleDeployDescrip
       : null
 
     def migCreateOperation
+    def instanceGroupManager = new InstanceGroupManager()
+        .setName(serverGroupName)
+        .setBaseInstanceName(serverGroupName)
+        .setInstanceTemplate(instanceTemplateUrl)
+        .setTargetSize(description.targetSize)
+        .setTargetPools(targetPools)
+        .setAutoHealingPolicies(autoHealingPolicy)
+
+    def hasBackendServices = description.instanceMetadata &&
+        description.instanceMetadata.containsKey(GoogleServerGroup.View.BACKEND_SERVICE_NAMES)
+
+    if (hasBackendServices && (description?.loadBalancingPolicy || description?.source?.serverGroupName))  {
+      NamedPort namedPort = null
+      def sourceGroupName = description?.source?.serverGroupName
+      // Note: this favors the explicitly specified load balancing policy over the source server group.
+      if (sourceGroupName && !description?.loadBalancingPolicy) {
+        def sourceServerGroup = googleClusterProvider.getServerGroup(description.accountName, description.source.region, sourceGroupName)
+        if (!sourceServerGroup) {
+          log.warn("Could not locate source server group ${sourceGroupName} to update named port.")
+        }
+        namedPort = new NamedPort(
+            name: GoogleHttpLoadBalancingPolicy.HTTP_PORT_NAME,
```

Comment:
Sgtm

On Jul 29, 2016 6:40 PM, "Jacob Kiefer" <notifications@github.com> wrote:

> In
> clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/handlers/BasicGoogleDeployHandler.groovy
> <https://github.com/spinnaker/clouddriver/pull/835#discussion_r72870836>:
>
> > +        .setAutoHealingPolicies(autoHealingPolicy)
> > +
> > +    def hasBackendServices = description.instanceMetadata &&
> > +        description.instanceMetadata.containsKey(GoogleServerGroup.View.BACKEND_SERVICE_NAMES)
> > +
> > +    if (hasBackendServices && (description?.loadBalancingPolicy || description?.source?.serverGroupName))  {
> > +      NamedPort namedPort = null
> > +      def sourceGroupName = description?.source?.serverGroupName
> > +      // Note: this favors the explicitly specified load balancing policy over the source server group.
> > +      if (sourceGroupName && !description?.loadBalancingPolicy) {
> > +        def sourceServerGroup = googleClusterProvider.getServerGroup(description.accountName, description.source.region, sourceGroupName)
> > +        if (!sourceServerGroup) {
> > +          log.warn("Could not locate source server group ${sourceGroupName} to update named port.")
> > +        }
> > +        namedPort = new NamedPort(
> > +            name: GoogleHttpLoadBalancingPolicy.HTTP_PORT_NAME,
>
> Each backend service only knows about one named port, so unless you're
> running a monolith with everything on one machine and routing to different
> ports on the same machine from multiple backend services, I think it makes
> sense to stick with the default name. If needed, we can add that in v2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spinnaker/clouddriver/pull/835/files/7480e3544570499fe66dc5daec784d694147ac61#r72870836>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIC_1vaXF0eX9UAYoCI0AXN7QqjeIiAsks5qaoF1gaJpZM4JYk8j>
> .
>